common/reflect_resource.go

```diff
@@ -633,36 +661,51 @@ func DiffToStructPointer(d attributeGetter, scm map[string]*schema.Schema, resul
 		panic(fmt.Errorf("pointer is expected, but got %s: %#v", reflectKind(rk), result))
 	}
 	rv = rv.Elem()
-	err := readReflectValueFromData([]string{}, d, rv, scm)
+	aliases := getAliasesMapFromStruct(result)
+	err := readReflectValueFromData([]string{}, d, rv, scm, aliases)
 	if err != nil {
 		panic(err)
 	}
 }
 
 // DataToStructPointer reads resource data with given schema onto result pointer. Panics.
 func DataToStructPointer(d *schema.ResourceData, scm map[string]*schema.Schema, result any) {
+	aliases := getAliasesMapFromStruct(result)
 	rv := reflect.ValueOf(result)
 	rk := rv.Kind()
 	if rk != reflect.Ptr {
 		panic(fmt.Errorf("pointer is expected, but got %s: %#v", reflectKind(rk), result))
 	}
 	rv = rv.Elem()
-	err := readReflectValueFromData([]string{}, d, rv, scm)
+	err := readReflectValueFromData([]string{}, d, rv, scm, aliases)
 	if err != nil {
 		panic(err)
 	}
 }
 
 // DataToReflectValue reads reflect value from data
 func DataToReflectValue(d *schema.ResourceData, s map[string]*schema.Schema, rv reflect.Value) error {
-	return readReflectValueFromData([]string{}, d, rv, s)
+	// TODO: Pass in the right aliases map.
+	return readReflectValueFromData([]string{}, d, rv, s, map[string]string{})
+}
+
+func getAliasesMapFromStruct(s any) map[string]string {
+	var aliases map[string]string
+	if v, ok := s.(ResourceProvider); ok {
+		aliases = v.Aliases()
+	} else {
+		aliases = map[string]string{}
+	}
+	return aliases
```

Comment:
Cleaner.
```suggestion
	if v, ok := s.(ResourceProvider); ok {
		return v.Aliases()
	}
	return map[string]string{}
```

Also, let's document that this doesn't return aliases defined on `tf` tags.